File: t/CS/RasterizeDocumentDemo/Tools/ValueNameItem.cs
// *************************************************************
// Copyright (c) 1991-2019 LEAD Technologies, Inc.              
// All Rights Reserved.                                         
// *************************************************************
using System;
using System.Collections.Generic;
using System.Text;

namespace RasterizeDocumentDemo.Tools
{
   // Used as an item for listbox and combobox controls
   class ValueNameItem<T>
   {
      private T _value;

      // The item value
      public T Value
      {
         get { return _value; }
         set { _value = value; }
      }

      // The item name
      private string _name;

      public string Name
      {
         get { return _name; }
         set { _name = value; }
      }

      public ValueNameItem(T value, string name)
      {
         _value = value;
         _name = name;
      }

      public override string ToString()
      {
         return _name;
      }

      // Return the selected item from a list
      public static object SelectItem(T value, System.Collections.IEnumerable items)
      {
         foreach(ValueNameItem<T> item in items)
         {
            if(item.Value.Equals(value))
               return item;
         }

         return null;
      }

      public static T GetSelectedItem(object item)
      {
         return (item as ValueNameItem<T>).Value;
      }
   }
}

File: t/VB/RasterizeDocumentDemo/Tools/ValueNameItem.vb
' *************************************************************
' Copyright (c) 1991-2019 LEAD Technologies, Inc.              
' All Rights Reserved.                                         
' *************************************************************
' Used as an item for listbox and combobox controls
Class ValueNameItem(Of T)
   Private _value As T

   ' The item value
   Public Property Value() As T
      Get
         Return _value
      End Get
      Set(ByVal value As T)
         _value = value
      End Set
   End Property

   ' The item name
   Private _name As String

   Public Property Name() As String
      Get
         Return _name
      End Get
      Set(ByVal value As String)
         _name = value
      End Set
   End Property

   Public Sub New(ByVal value As T, ByVal name As String)
      _value = value
      _name = name
   End Sub

   Public Overrides Function ToString() As String
      Return _name
   End Function

   ' Return the selected item from a list
   Public Shared Function SelectItem(ByVal value As T, ByVal items As System.Collections.IEnumerable) As Object
      For Each item As ValueNameItem(Of T) In items
         If item.Value.Equals(value) Then
            Return item
         End If
      Next

      Return Nothing
   End Function

   Public Shared Function GetSelectedItem(ByVal item As Object) As T
      Return CType(item, ValueNameItem(Of T)).Value
   End Function
End Class
